com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/VertMesh.hlsl

```diff
@@ -177,7 +177,13 @@ VaryingsMeshToPS VertMeshTesselation(VaryingsMeshToDS input)
     output.positionCS = TransformWorldToHClip(input.positionRWS);
 
 #if defined(SHADERPASS) && (SHADERPASS == SHADERPASS_FULLSCREEN_DEBUG)
+#ifdef SHADER_API_METAL
+    // There is a bug with generated shader on metal when accessing the ShaderVariablesDebugDisplay cbuffer in this shader stage.
```

Comment:
I am not sure to understand the workaround, why not doing 

#ifndef SHADER_API_METAL 
#if defined(SHADERPASS) && (SHADERPASS == SHADERPASS_FULLSCREEN_DEBUG)
if (_DebugFullScreenMode == FULLSCREENDEBUGMODE_VERTEX_DENSITY)
 IncrementVertexDensityCounter(output.positionCS);
#endif
#"endif?

or simpler, inside IncrementVertexDensityCounter you can do the test on SHADER_API_METAL

Reply:
This bug is for vertex density, not quad overdraw, and it should work on mac. so with what I did it is supposed to work also on mac.